test/upgrade/prober/wathola/event/services.go

```diff
@@ -94,21 +95,35 @@ func (f *finishedStore) RegisterFinished(finished *Finished) {
 			f.received+1)
 	}
 	f.received++
-	f.count = finished.Count
-	log.Infof("finish event received, expecting %d event ware propagated", finished.Count)
+	f.eventsSent = finished.EventsSent
+	f.totalRequests = finished.TotalRequests
+	log.Infof("finish event received, expecting %d event ware propagated", finished.EventsSent)
 	d := config.Instance.Receiver.Teardown.Duration
 	log.Infof("waiting additional %v to be sure all events came", d)
 	time.Sleep(d)
 	receivedEvents := f.steps.Count()
-	if receivedEvents != finished.Count {
+	if receivedEvents != finished.EventsSent {
 		f.errors.throwUnexpected("expecting to have %v unique events received, "+
-			"but received %v unique events", finished.Count, receivedEvents)
+			"but received %v unique events", finished.EventsSent, receivedEvents)
 		f.reportViolations(finished)
 		f.errors.state = Failed
 	} else {
 		log.Infof("properly received %d unique events", receivedEvents)
 		f.errors.state = Success
 	}
+	// check down time
+	for _, unavailablePeriod := range finished.UnavailablePeriods {
+		if unavailablePeriod > config.Instance.Receiver.ErrorCfg.UnavailablePeriodToReport {
+			// TODO: decide how to do this properly
+			f.errors.throwUnexpected("actual unavailable period %v is over down time limit of %v", unavailablePeriod, config.Instance.Receiver.ErrorCfg.UnavailablePeriodToReport)
```

Comment:
Looks like we will have to use another error type, thoughts? @cardil

Reply:
Yeah. I think we should introduce new error type, so not to rely on string matching, but on types.

Reply:
Done as suggested.